File: 2 SEMESTER/Object Oriented Programming/Projects/PROJECT2_JAVA/src/Organisms/Plants/Guarana.java
package Organisms.Plants;

import Game.World;
import Organisms.Organism;

import java.awt.*;
import java.io.Serializable;

public class Guarana extends Plant implements Serializable {
    public Guarana(World world, int x, int y)
    {
        super("R", "Guarana", Color.pink, 0, 0, 1, world,x,y);
    }
    @Override
    public Guarana getChild(int x, int y)
    {
        return new Guarana(world,x,y);
    }
    @Override
    public void Collision(Organism attacker)
    {
        attacker.setStrength(attacker.getStrength()+3);
        super.Collision(attacker);
    }
}

File: 2 SEMESTER/Object Oriented Programming/Projects/PROJECT3_PYTHON/Organisms/Plants/guarana.py
from Organisms.Plants.plant import Plant


class Guarana(Plant):
    def __init__(self, position_x, position_y, world):
        super().__init__(position_x, position_y, "Guarana", 'G', 0, 0, world, 0, (255, 192, 203))

    def getchild(self, newposition):
        return Guarana(newposition[0], newposition[1], self.world)

    def planteffect(self, attacker):
        attacker.strength += 5
        self.world.logs.append(self.tostring() + " wzmacnia " + attacker.tostring())